### ISSUE TITLE
Bowtie symbol isn't a valid character.

### ISSUE THREAD

**Issue by rofinn** [JuliaLang/julia#23820]
```julia
julia> ⋈ = 1
ERROR: syntax: invalid character "⋈"
```
FWIW, `\bowtie` tab completed on the REPL.

**Comment by JeffBezanson** [comment]
Ah, one of those category Sm characters. They have a range of meanings so we need to whitelist them. Should this be an operator or just an identifier character?

**Comment by stevengj** [comment]
Yes, [⋈ (U+22c8)](http://www.fileformat.info/info/unicode/char/22c8/index.htm) was one of the characters that got left out because it wasn't immediately clear if it should be an operator or an identifier.

It seems like it is mainly used as a [relation symbol](http://web.ift.uib.no/Teori/KURS/WRK/TeX/sym3.html). Most of these [relational algebra symbols](https://en.wikipedia.org/wiki/Relational_algebra) were added in #8036.    Confusingly, the [join symbol ⨝ (U+2a1d)](http://www.fileformat.info/info/unicode/char/2a1d/index.htm) is already supported as an infix operator, and looks nearly identical to the "bowtie".

One option would be to add a mapping to our [custom Julia Unicode normalization](https://github.com/JuliaLang/julia/pull/1946) that simply parses ⋈ (U+22c8) as equivalent to ⨝ (U+2a1d).

**Comment by rofinn** [comment]
I would like to use it as an infix operator for natural join between 2 relations, but I'd be fine if it was just a valid function name.

**Comment by rofinn** [comment]
Oh, I didn't realize they were different unicode symbols. I'm fine using the join symbol, but `\join` wasn't tab completing on the REPL in 0.6 or 0.7 (5 days old).

**Comment by StefanKarpinski** [comment]
It's capital `\Join` not `\join` (I have no idea why).

**Comment by StefanKarpinski** [comment]
Normalizing these seems sane since they like exactly alike.

**Comment by stevengj** [comment]
The reason it was `\Join` is because it is that way in LaTeX, and we were hesitant about innovating in LaTeX names in #8036.   I agree that adding the lower-case tab completion, along with the normalization, makes sense.

**Comment by stevengj** [comment]
One oddity is that `\Join` seems to be treated as a fullwith character, while bowtie is treated as narrow:
```jl
julia> charwidth('⋈') # bowtie
1

julia> charwidth('⨝')# join
2
```
(`\Join` displays oddly in the REPL too, similar to #3721, since the terminal also seems to be confused about its character width.)

**Comment by JeffBezanson** [comment]
My terminal and/or font seems to have the widths swapped --- bowtie takes up 2 columns (though the terminal doesn't know it) and join takes up 1.

**Comment by iamed2** [comment]
My terminal handles `\Join` correctly but Julia doesn't

**Comment by rofinn** [comment]
Would there be any issue with making all the unicode substitutions case-insensitive? Is there an example where the casing would matter?

**Comment by stevengj** [comment]
@rofinn, since the same tab-substitution list is now supported by several editors, and we can't control the case sensitivity of editor tab substitution without providing an explicit list of all possibilities (`\Join`, `\JOiN`, …), I think it's better for consistency's sake to just a support a limited set of variations. (e.g. `\Join` and `\join`). 

See also the discussion in #21646 about rationalizing the latex completions.

**Comment by StefanKarpinski** [comment]
IIRC, there are examples where case matters in LaTeX escapes.

**Comment by iamed2** [comment]
Example: 

`\delta`: δ
`\Delta`: Δ

**Comment by rofinn** [comment]
Alright, I forgot about `\delta` vs `\Delta`.
